### PR TITLE
refine: shuffle column order before execute function ut

### LINKED ISSUE #5596: enhance ut framework to shuffle column order before execution
## Enhancement

Current `executeFunction` in function ut framwork manually constructs columns with the input order. Therefore, while executing ut, the logical offset and physical offset of each column is the same, which usually is not the case in real environment. 

```cpp
ColumnWithTypeAndName executeFunction(
    Context & context,
    const String & func_name,
    const ColumnsWithTypeAndName & columns,
    const TiDB::TiDBCollatorPtr & collator,
    bool raw_function_test)
{
    ColumnNumbers argument_column_numbers;
    for (size_t i = 0; i < columns.size(); ++i)
        argument_column_numbers.push_back(i);
    return executeFunction(context, func_name, argument_column_numbers, columns, collator, raw_function_test);
}
```

A relevant bug failed to be discovered by ut can be found at #5592.

Therefore, it would be better if we can introduce some randomness to the framework and change the order of input columns before doing execution.

### PR DESCRIPTION
**Pull request by Willendless**
### What problem does this PR solve?

Issue Number: close #5596 

Problem Summary:

### What is changed and how it works?

Shuffle input columns before execution in function unit tests.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 1]
```diff
@@ -293,7 +305,15 @@ ColumnWithTypeAndName executeFunction(
     ColumnNumbers argument_column_numbers;
     for (size_t i = 0; i < columns.size(); ++i)
         argument_column_numbers.push_back(i);
-    return executeFunction(context, func_name, argument_column_numbers, columns, collator, raw_function_test);
+
+    /// shuffle input columns to assure function correctly use physical offsets instead of logical offsets
```
Comment (ywqzzy) [inline dbms/src/TestUtils/FunctionTestUtils.cpp:309]: explain why we need the shuffle process.

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Pull request merged**